web/src/components/dashboard/SidebarCardFundingNudge.test.tsx

```diff
@@ -1,46 +1,57 @@
 import { SidebarCardFundingNudge } from "@/components/dashboard/SidebarCardFundingNudge";
 import { getMergedConfig } from "@/contexts/configContext";
+import * as api from "@/lib/api-client/apiClient";
 import { SidebarCardContent } from "@/lib/types/types";
-import analytics from "@/lib/utils/analytics";
-import { generateSidebarCardContent } from "@/test/factories";
+import { generateEmptyFormationData, generateSidebarCardContent } from "@/test/factories";
 import { selectDropdownByValue } from "@/test/helpers/helpers-testing-library-selectors";
+import { randomElementFromArray } from "@/test/helpers/helpers-utilities";
 import { useMockRouter } from "@/test/mock/mockRouter";
 import {
   currentBusiness,
   setupStatefulUserDataContext,
   userDataWasNotUpdated,
   WithStatefulUserData,
 } from "@/test/mock/withStatefulUserData";
-import { OperatingPhaseId } from "@businessnjgovnavigator/shared/";
+import { getCurrentBusiness, OperatingPhaseId, OperatingPhases } from "@businessnjgovnavigator/shared";
 import { SIDEBAR_CARDS } from "@businessnjgovnavigator/shared/domain-logic/sidebarCards";
 import {
   generateBusiness,
+  generateFormationData,
+  generateGetFilingResponse,
   generatePreferences,
   generateProfileData,
+  generateTaxFilingData,
   generateUserDataForBusiness,
+  randomIndustry,
+  randomLegalStructure,
+  randomPublicFilingLegalType,
 } from "@businessnjgovnavigator/shared/test";
 import { Business } from "@businessnjgovnavigator/shared/userData";
-import { fireEvent, render, screen } from "@testing-library/react";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
 
 jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
 jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+jest.mock("@/lib/api-client/apiClient", () => ({
+  postTaxFilingsOnboarding: jest.fn(),
+  postTaxFilingsLookup: jest.fn(),
+}));
+
 jest.mock("next/router", () => ({ useRouter: jest.fn() }));
-jest.mock("@/lib/utils/analytics", () => setupMockAnalytics());
-
-const mockAnalytics = analytics as jest.Mocked<typeof analytics>;
-function setupMockAnalytics(): typeof analytics {
-  return {
-    ...jest.requireActual("@/lib/utils/analytics").default,
-    event: {
-      ...jest.requireActual("@/lib/utils/analytics").default.event,
-      show_me_funding_opportunities: {
-        click: {
-          show_me_funding_opportunities: jest.fn(),
-        },
+
+const mockApi = api as jest.Mocked<typeof api>;
+
+const mockApiResponse = (userData: UserData, overrides: Partial<Business>): void => {
```

Comment:
Import for `UserData`, `FormationData`, and `FormationLegalType` are missing. This is probably one of the things the linter is yelling about.

This is something that jumped out in my IDE. We can take a look at settings if these aren't popping up for you.